iceoryx2-bb/posix/src/periodic_timer.rs

```diff
@@ -0,0 +1,235 @@
+        self.clock_type = value;
+        self
+    }
+
+    /// Creates a new [`PeriodicTimer`]
+    pub fn create(self) -> Result<PeriodicTimer, TimeError> {
+        let start_time = fail!(from "PeriodicTimer::new()", when Time::now_with_clock(self.clock_type),
+                                "Failed to create PeriodicTimer since the current time could not be acquired.");
+        let start_time = start_time.as_duration().as_nanos();
+
+        Ok(PeriodicTimer {
+            timeouts: RefCell::new(vec![]),
+            id_count: IoxAtomicU64::new(0),
+            clock_type: self.clock_type,
+            last_iteration: RefCell::new(start_time),
+        })
+    }
+}
+
+#[derive(Debug)]
+struct Attachment {
+    index: u64,
+    period: u128,
+    start_time: u128,
+}
+
+impl Attachment {
+    fn new(index: u64, period: u128, clock_type: ClockType) -> Result<Self, TimeError> {
+        let start_time = fail!(from "Attachment::new()", when Time::now_with_clock(clock_type),
+                                "Failed to create PeriodicTimer attachment since the current time could not be acquired.");
+        let start_time = start_time.as_duration().as_nanos();
+
+        Ok(Self {
+            index,
+            period,
+            start_time,
+        })
+    }
+
+    fn reset(&mut self, clock_type: ClockType) -> Result<(), TimeError> {
+        let start_time = fail!(from "Attachment::new()", when Time::now_with_clock(clock_type),
+                                "Failed to reset PeriodicTimer attachment since the current time could not be acquired.");
+        self.start_time = start_time.as_duration().as_nanos();
+        Ok(())
+    }
+}
+
+/// The [`PeriodicTimer`] allows the user to attach multiple periodic timers with
+/// [`PeriodicTimer::add()`], to wait on them by acquiring the waiting time to the next timer
+/// with [`PeriodicTimer::next_iteration()`] and to acquire all missed timers via
+/// [`PeriodicTimer::missed_timers()`].
+#[derive(Debug)]
+pub struct PeriodicTimer {
+    timeouts: RefCell<Vec<Attachment>>,
+    id_count: IoxAtomicU64,
+    last_iteration: RefCell<u128>,
+
+    clock_type: ClockType,
+}
+
+impl PeriodicTimer {
+    /// Adds a cyclic timeout to the [`PeriodicTimer`] and returns an [`PeriodicTimerGuard`] to
+    /// identify the attachment uniquely.
+    /// [`PeriodicTimer::next_iteration()`] will schedule the timings in a way that the attached
+    /// timeout is considered cyclicly.
+    pub fn cyclic(&self, timeout: Duration) -> Result<PeriodicTimerGuard, TimeError> {
+        let current_idx = self.id_count.load(Ordering::Relaxed);
+        self.timeouts.borrow_mut().push(Attachment::new(
+            current_idx,
+            timeout.as_nanos(),
+            self.clock_type,
+        )?);
+        self.id_count.fetch_add(1, Ordering::Relaxed);
+
+        Ok(PeriodicTimerGuard {
+            periodic_timer: self,
+            index: current_idx,
+        })
+    }
+
+    fn remove(&self, index: u64) {
```

Comment:
Can we assume that `Attachment` (or `Interval`) instances are available to the caller, and they can pass this instead of the index? Or is there a way for a caller to know a specific index to be removed without having created the `Attachment` ?

Reply:
This is a private method only used by the `PeriodicTimerGuard` to remove itself when it goes out of scope.